MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -2317,6 +2320,10 @@ static void signalAndUntrack(const MVKSwapchainSignaler& signaler) {
 void MVKImageView::destroy() {
 	detachMemory();
 	MVKVulkanAPIDeviceObject::destroy();
+    if (_image) {
+        _image->release();
+        _image = NULL;
+    }
```

Comment:
So, the reason that `MVKBaseObject::destroy()` exists in the first place is that Metal could still indirectly reference one of these objects even after the Vulkan spec says it's OK to destroy them, through a completion block. These objects are `retain()`ed so that this case doesn't crash. The objects should still be destroyed after the completion blocks have finished running.

Releasing the image here could cause a crash later on. Why is this here? Are you seeing a case where the `MVKImage` and `MVKImageView` objects are getting leaked?

Reply:
> Are you seeing a case where the MVKImage and MVKImageView objects are getting leaked?

I am, but only on some older hardware, and not in all cases. I only notice the leak after/as a result of 188a21ad33df4705ea4050f0116939bee097c7fe, and if I revert that I no longer see it. It's been sort of hard for me to narrow down the specific case that's causing the leak.

I understand what you're saying though. I'll see if I can go through and figure out which objects are being destroy()'d but not having their destructors called, and where that destroy() is coming from.

Reply:
I've updated my PR based on my investigation that I've documented in #2341 ; basically `MVKImageMemoryBinding::bindMemoryAddress()` ends up getting called from the `MVKDeviceMemory` destructor with a `nullptr`, but because `_image->_isLinearForAtomics` is true it allocates a buffer that never gets freed.